### PR TITLE
MEN-6580: Implement deployment logging in deployments

### PR DESCRIPTION
**Pull request by kacf**
Plus some supporting commits and fixes.

### CODE REVIEW

File: common/state_machine.hpp [hunk 1]
```diff
@@ -228,7 +228,7 @@ class StateMachineRunner : virtual public EventPoster<EventType> {
 
 		auto cancelled = cancelled_;
 		event_loop_->Post([cancelled, this]() {
-			if (!*cancelled_) {
+			if (!*cancelled) {
```
Comment (kacf) [inline common/state_machine.hpp:231]: @lluiscampos: Very relevant for what you're doing.

File: mender-update/daemon/context.cpp [hunk 2]
```diff
@@ -521,6 +521,28 @@ expected::ExpectedBool Context::LoadDeploymentStateData(StateData &state_data) {
 	}
 }
 
+void BeginDeploymentLogging(Context &ctx) {
```
Comment (vpodzime) [inline mender-update/daemon/context.cpp:524]: Why not make this a member function of `Context`?

**Review by vpodzime — APPROVED** [review]
Looks good to me otherwise.

File: mender-update/daemon/state_test.cpp [hunk 3]
```diff
@@ -3140,6 +3140,13 @@ TEST_P(StateDeathTest, StateTransitionsTest) {
 
 	string status_log_path = path::Join(tmpdir.Path(), "status.log");
 
+	{
+		// We don't care about existence, only content, so pre-create these files in order
+		// to make FileContains* functions happy.
+		ofstream state_log(state_log_path);
```
Comment (oleorhagen) [inline mender-update/daemon/state_test.cpp:3158]: ?

File: mender-update/daemon/state_machine/state_machine.cpp [hunk 4]
```diff
@@ -291,6 +298,8 @@ error::Error StateMachine::Run() {
 		return err;
 	}
 
+	log::Info("Running Mender client " + conf::kMenderVersion);
```
Comment (oleorhagen) [inline mender-update/daemon/state_machine/state_machine.cpp:301]: Q: Wrong commit?

**Review by oleorhagen — APPROVED** [review]

**Review by lluiscampos — APPROVED** [review]

File: mender-update/daemon/state_test.cpp (hunk 3 above)
Comment (kacf) [reply mender-update/daemon/state_test.cpp:3158]: What's the question?

File: mender-update/daemon/state_machine/state_machine.cpp (hunk 4 above)
Comment (kacf) [reply mender-update/daemon/state_machine/state_machine.cpp:301]: No, it's correct. We want each deployment log, as well as after every reboot, to start with the version, for debugging reasons in the field, like we do [in the Golang client](https://github.com/mendersoftware/mender/blob/3.5.1/app/deployment_logger.go#L144).

File: mender-update/daemon/state_test.cpp (hunk 3 above)
Comment (oleorhagen) [reply mender-update/daemon/state_test.cpp:3158]: Never mind

**Pull request merged**